src/serialization/html-serializer.tsx

```diff
@@ -1,78 +1,109 @@
-import { HtmlSerializationRule } from "../plugins/html-serializable-plugin";
-import { ColorPlugin } from "../plugins/color-plugin";
-import { CoreBlocksPlugin } from "../plugins/core-blocks-plugin";
-import { CoreInlinesPlugin } from "../plugins/core-inlines-plugin";
-import { CoreMarksPlugin } from "../plugins/core-marks-plugin";
-import { ImagePlugin } from "../plugins/image-plugin";
-import { LinkPlugin } from "../plugins/link-plugin";
-import { ListPlugin } from "../plugins/list-plugin";
-import { TablePlugin } from "../plugins/table-plugin";
+import { Element, elementTypeMap } from '../slate-editor/element';
+import React from 'react';
+import { isCustomText } from '../slate-editor/slate-utils';
+import { SerializingContext } from '../hooks/use-serializing';
+import { Descendant } from 'slate';
 
-// A modified version of the TEXT_RULE from the slate-html-serializer with
-// special handling for &nbsp;
-const TEXT_RULE: HtmlSerializationRule = {
-  deserialize(el) {
-    if (el.tagName && el.tagName.toLowerCase() === 'br') {
-      return {
-        object: 'text',
-        text: '\n',
-        marks: [],
-      };
-    }
 
-    if (el.nodeName === '#text') {
-      if (el.nodeValue?.match(/<!--.*?-->/)) return;
+//FIXME: TYPES!
+const deserialize = (el : any, markAttributes = {}) : any => {
+  if (el.nodeType === Node.TEXT_NODE) {
+    return jsx('text', markAttributes, el.textContent);
+  } else if (el.nodeType !== Node.ELEMENT_NODE) {
+    return null;
+  }
+
+  const nodeAttributes:any = { ...markAttributes }; // FIXME: type
+
+  const textMark = markNodeMap[el.nodeName.toLowerCase()];
+  if (textMark) {
+    nodeAttributes[textMark] = true;
+  }
 
-      return {
-        object: 'text',
-        text: el.nodeValue,
-        marks: [],
-      };
-    }
-  },
+  const children = Array.from(el.childNodes)
+    .map(node => deserialize(node, nodeAttributes))
+    .flat();
 
-  serialize(obj, children) {
-    if (obj.object === 'string') {
-      return children.split('\n').reduce((array, text, i) => {
-        if (i !== 0) array.push(<br key={i} />);
-        // encode non-breaking spaces (for visibility)
-        array.push(text.replace(/\u00A0/g, "&nbsp;"));
-        return array;
-      }, [] as ReactNode[]);
-    }
-  },
+  if (children.length === 0) {
+    children.push(jsx('text', nodeAttributes, ''));
+  }
 
-  postSerialize: function(html) {
-    // After encoding non-breaking spaces in the TEXT_RULE above,
-    // renderToStaticMarkup() re-escapes so we have to unescape.
-    return html.replace(/&amp;nbsp;/g, "&nbsp;");
+  // This is assuming there's 1:1 mapping of tag to elements which is probably not true...
+  const elementTag = elementTypeMap[el.nodeName.toLowerCase()];
+  if (elementTag) {
+    const atts = el.getAttributeNames();
+    const obj = {};
+    obj.type = elementTag;
+    // FIXME:
+    // Copy attributes from html tag to jsx object. 
+    // This isn't necessarily a 1:1 mapping so this only kinda works.
```

Comment:
This assumption is likely why the test is broken (see comment below).  hypothetical (although not true) example: if a link's url were called "url"  in the slate representation, it would deserialize as `<a url="<whatever">` instead of `<a href="<whatever">`